index.js

```diff
@@ -28,6 +28,13 @@ function File(file) {
   this.contents = file.contents || null;
 
   this._isVinyl = true;
+
+  // Set custom properties
+  Object.keys(file).forEach(function(key) {
+    if (this.constructor.isCustomProp(key)) {
```

Comment:
It'll always be a function, since `extends` inherits static methods as well. It'd be the same as checking if `file.isBuffer()` is a function. Of course it is if it inherits from Vinyl :)